### PR TITLE
[query] refacotr: Catalog: make get_databases() async

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

##### [query] refacotr: Catalog: make get_databases() async
- fix: #2593
- fix: #2591

- As discussed in #2492  convert methods to async if possible.

There are 2 async-to-sync conversion left in DatabendQueryContext:

`Catalog::get_database()` and `Catalog::get_table()`.

## Related Issues

- #2030

### CODE REVIEW

**Review by dantengsky — APPROVED** [review]
lgtm

File: query/src/sessions/context.rs [hunk 1]
```diff
@@ -179,7 +181,14 @@ impl DatabendQueryContext {
     }
 
     pub fn set_current_database(&self, new_database_name: String) -> Result<()> {
-        match self.get_catalog().get_database(new_database_name.as_str()) {
+        let rt = self.shared.try_get_runtime()?;
+        let cata = self.get_catalog();
+        let db_name = new_database_name.clone();
+
+        let res = (async move { cata.get_database(&db_name).await })
+            .wait_in(&rt, Some(Duration::from_millis(5000)))?;
```
Comment (ariesdevil) [inline query/src/sessions/context.rs:189]: I think `5000` should be a constant or feed it in the config file.

File: query/src/sessions/context_shared.rs [hunk 2]
```diff
@@ -122,7 +124,11 @@ impl DatabendQueryContextShared {
             Entry::Occupied(entry) => entry.get().clone(),
             Entry::Vacant(entry) => {
                 let catalog = self.get_catalog();
-                let t = catalog.get_table(database, table)?;
+                let rt = self.try_get_runtime()?;
+                let database = database.to_string();
+                let table = table.to_string();
+                let t = (async move { catalog.get_table(&database, &table).await })
+                    .wait_in(&rt, Some(Duration::from_millis(5000)))??;
```
Comment (ariesdevil) [inline query/src/sessions/context_shared.rs:131]: Same as above.

File: query/src/sessions/context.rs (hunk 1 above)
Comment (BohuTANG) [reply query/src/sessions/context.rs:189]: Guess `get_database` should async soon :thinking:

**Pull request merged**